content/posts/2019-05-02-apollo-client-in-practice.md

```diff
@@ -19,15 +19,15 @@ computations in the bottom components (e.g. formatting in cells). Because there
 re-rendered quite often, this approach is quite intensive on resources.
 
 Besides that, the application had obvious state synchronization problems. When moving between views it was not 
-maintaining the same state of selected rows. Even though Redux is used to store application state and communicate it 
-between components, it is not used consistently. There are still plenty of React class components that store some parts of the state locally. 
+maintaining the same state of selected rows. Even though Redux was used to store application state and communicate it 
+between components, it was not used consistently. There were still plenty of React class components that stored some parts of the state locally. 
 
-To summarize, there are two issues that need to be solved:
+To summarize, there were two issues that needed to be solved:
 
 1. Reduce re-rendering to improve performance
```

Comment:
Technically, these are two solutions. The issues would appear to be:

Poor performance due to needless re-rendering
Loss of state due to fractured application state